### PR TITLE
Feature/update docusaurus

### PR DESCRIPTION
**Pull request by dingo-d**
This is a WIP built on top of #133. 

The docusaurus has been updated to beta-15 version. 

All the errors during the build process have been fixed.

Blog needs to be styled, and mobile menu needs to be restyled.

### CODE REVIEW

File: website/src/theme/components/blog-layout/blog-layout.scss [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// This file has multiple nestings because Docusaurus adds its custom selectors that he overrides in build phase and doesn't take into account your overrides.
+// Themes are not mend to be overriden whole but only partial.
+
+$blog-list-page: (
+main-wrapper: map-get-strict($base-containers, default),
+gutter: $base-gutter-size
+);
```
Comment (dingo-d) [inline website/src/theme/components/blog-layout/blog-layout.scss:7]: The indentation is off here

**Comment by dingo-d** [comment]
@iobrado can you check the mobile menu? Currently, it's not looking nicely (it was changed completely in the beta 15 release).

**Review by gabriel-glo — APPROVED** [review]

**Comment by iobrado** [comment]
I've managed to fix mobile menu and I've also updated hover effects on blog archive since it only added underline without any transition effects. Here are a few videos showing the changes.

https://user-images.githubusercontent.com/23059501/155084188-62cb00f6-a9f5-4efa-a241-e44809231fb6.mov


https://user-images.githubusercontent.com/23059501/155084205-4cdc9148-e49c-42c6-8cfe-ef583064933e.mov

**Comment by dingo-d** [comment]
Love how you've styled the mobile menu, the one thing I'm not really thrilled about is the over effect on the blog title. Could you make it so it behaves like a menu? Default is green/black and on hover, it's red IIRC?

**Comment by iobrado** [comment]
@dingo-d how about this? And what about author name and tags? Can they stay as-is or do you have any suggestions for styling those?

https://user-images.githubusercontent.com/23059501/155110640-7325c9a9-5bf7-47c6-8db3-90874a86cb47.mov

**Comment by dingo-d** [comment]
Much better :cheffs_kiss: 😄

**Review by iruzevic — APPROVED** [review]

File: website/docusaurus.config.js [hunk 2]
```diff
@@ -73,23 +80,29 @@ module.exports = {
...
         },
+        gtag: {
+          trackingID: 'GTM-P5GG5DH',
+          anonymizeIP: true,
+        },
         theme: {
           customCss: require.resolve('./src/scss/application.scss'),
         },
+        blog: {
+          blogTitle: 'Tutorials and articles about Eightshift development kit',
```
Comment (goranalkovic-infinum) [inline website/docusaurus.config.js:91]:
```suggestion
          blogTitle: 'Tutorials and articles related to the Eightshift development kit',
```

File: website/src/theme/components/blog-post-page/blog-post-page.scss [hunk 3]
```diff
@@ -0,0 +1,65 @@
+// This file has multiple nestings because Docusaurus adds its custom selectors that he overrides in build phase and doesn't take into account your overrides.
+// Themes are not mend to be overriden whole but only partial.
```
Comment (goranalkovic-infinum) [inline website/src/theme/components/blog-post-page/blog-post-page.scss:2]:
```sugggestion
// Themes are not meant to be overridden as a whole, only partially.
```

File: website/src/theme/components/blog-post-page/blog-post-page.scss [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+				padding: 0 !important;
+			}
+
+			li {
+				margin-bottom: 25px;
+			}
+		}
+
+		footer.docusaurus-mt-lg {
+			margin-top: 50px;
```
Comment (goranalkovic-infinum) [inline website/src/theme/components/blog-post-page/blog-post-page.scss:33]: Please use REMs

File: website/src/theme/components/navbar-sidebar/navbar-sidebar.scss [hunk 5]
```diff
@@ -1,7 +1,7 @@
 $navbar-sidebar: (
   link: (
     mobile: (
-      font-size: 38px,
+      font-size: 24px,
```
Comment (goranalkovic-infinum) [inline website/src/theme/components/navbar-sidebar/navbar-sidebar.scss:4]: Same as 🆙

File: website/src/theme/components/blog-sidebar/blog-sidebar.scss [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#__docusaurus .blog-list-page {
+    aside {
+        padding: 10px $base-gutter-size 20px !important;
```
Comment (goranalkovic-infinum) [inline website/src/theme/components/blog-sidebar/blog-sidebar.scss:3]: Same as 🆙

**Review by goranalkovic-infinum — COMMENTED** [review]
Left a couple of suggestions, nice job!

(Realized after a couple of files that there are a lot of places with `px` as units, so I didn't want to add too many comments)

**Comment by iobrado** [comment]
> (Realized after a couple of files that there are a lot of places with `px` as units, so I didn't want to add too many comments)

The reason for that is that I was following how the existing scss files were structured. I didn't want to make a mess that half files have pixels and other have REMs.

**Pull request merged**